www/class/centreonCustomView.class.php

```diff
@@ -185,6 +186,44 @@ public function checkPermission($viewId)
         return true;
     }
 
+    /**
+     * checkUserActions - Check wether or not the user
+     * can do the provided action
+     *
+     * @param  string $action
+     * @return bool
+     */
+    public function checkUserActions(string $action): bool
+    {
+        /* associate the called action with the toplogy page.
+         * Special behaviour for the deleteWidget action which does not have any possible configuration in ACL Menus.
+         * We do consider through that if user can create a widget then he also can delete it.
+         * Also "defaultEditMode" action has no link with ACL this action is authorized for all.
+         */
+        if ($action == 'defaultEditMode') {
+            return true;
+        }
+        $associativeActions = [
+            'edit' => '10301',
+            'share' => '10302',
+            'setPreferences' => '10303',
+            'addWidget' => '10304',
+            'deleteWidget' => '10304',
+            'setRotate' => '10305',
+            'deleteView' => '10306',
+            'add' => '10307',
+            'setDefault' => '10308'
```

Comment:
Could you move topology page numbers into separated constants ?
eg: `public const TOPOLOGY_PAGE_EDIT = '10301';`
You can keep this associative array, using declared constants

Reply:
Yes sure :+1:

Reply:
@kduret done :)